tests/modules/aclcheck.c

```diff
@@ -109,15 +113,31 @@ int rm_call_aclcheck_cmd(ValkeyModuleCtx *ctx, ValkeyModuleUser *user, ValkeyMod
     return VALKEYMODULE_OK;
 }
 
-int rm_call_aclcheck_cmd_default_user(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, int argc) {
-    ValkeyModuleString *user_name = ValkeyModule_GetCurrentUserName(ctx);
-    ValkeyModuleUser *user = ValkeyModule_GetModuleUserFromUserName(user_name);
+int rm_call_aclcheck_cmd(ValkeyModuleCtx *ctx, ValkeyModuleUser *user, ValkeyModuleString **argv, int argc) {
+    if (argc < 2) {
+        return ValkeyModule_WrongArity(ctx);
+    }
 
-    int res = rm_call_aclcheck_cmd(ctx, user, argv, argc);
+    /* Check that the command can be executed */
+    int ret = ValkeyModule_ACLCheckCommandPermissions(user, argv + 1, argc - 1);
+    if (ret != 0) {
+        ValkeyModule_ReplyWithError(ctx, "DENIED CMD");
+        /* Add entry to ACL log */
+        ValkeyModule_ACLAddLogEntry(ctx, user, argv[1], VALKEYMODULE_ACL_LOG_CMD);
+        return VALKEYMODULE_OK;
+    }
 
-    ValkeyModule_FreeModuleUser(user);
-    ValkeyModule_FreeString(ctx, user_name);
-    return res;
+    const char* cmd = ValkeyModule_StringPtrLen(argv[1], NULL);
+
+    ValkeyModuleCallReply* rep = ValkeyModule_Call(ctx, cmd, "v", argv + 2, (size_t)argc - 2);
+    if(!rep){
+        ValkeyModule_ReplyWithError(ctx, "NULL reply returned");
+    }else{
```

Comment:
Formatting nit. (We don't run clang-format on files in this directory?)

```suggestion
    if (!rep) {
        ValkeyModule_ReplyWithError(ctx, "NULL reply returned");
    } else {
```